### PR TITLE
Drop PHP 7, add PHP 8.2 support

### PR DESCRIPTION
**Pull request by glo82145**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### CODE REVIEW

File: .laminas-ci.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "ignore_php_platform_requirements": {
+      "8.2": true
+    }
+}
```
Comment (froschdesign) [inline .laminas-ci.json]: This should be avoided in the first step.

File: composer.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
         "sort-packages": true
     },
     "require": {
-        "php": "^7.3 || ~8.0.0 || ~8.1.0",
+        "php": ">=7.4, <8.3",
```
Comment (froschdesign) [inline composer.json]: Version 7.4 of PHP can be dropped because [number is 7 dead](https://www.php.net/supported-versions).

And please follow the previous version constraints `~8.0.0 || ~8.1.0 || ~8.2.0`.

**Review by froschdesign — CHANGES_REQUESTED** [review]

**Comment by glo82145** [comment]
@froschdesign Changes are done as suggested .. Please have a look

**Comment by fezfez** [comment]
> @froschdesign Changes are done as suggested .. Please have a look

you must run phpcbf

File: composer.json [hunk 3]
```diff
@@ -54,10 +54,12 @@
...
-        "test": "phpunit --colors=always",
-        "test-coverage": "phpunit --colors=always --coverage-clover clover.xml"
+        "test": "phpunit --colors=always --testsuite \"unit test\"",
+        "test-coverage": "phpunit --colors=always --coverage-clover clover.xml",
+        "test-integration": "phpunit --colors=always --testsuite \"integration test\"",
+        "upload-coverage": "coveralls -v"
     },
     "conflict": {
-        "zendframework/zend-file": "*"
+        "zendframework/zend-db": "*"
```
Comment (froschdesign) [inline composer.json]: Please undo these changes as the conflict is wrong and laminas-file does not contain integration tests. (laminas-db is not the correct reference here).

Comment (froschdesign) [reply composer.json]: Please add also the [`platform` configuration](https://getcomposer.org/doc/06-config.md#platform):

```json
"config": {
    "sort-packages": true,
    "platform": {
        "php": "8.0.99"
    }
}
```
Compare with: https://github.com/laminas/laminas-validator/blob/e0d2d90488d3e2427c1a3d325269529f605648a6/composer.json#L20-L22

**Comment by Ocramius** [comment]
I think it did a "latest" PHPCS installation due to:

```
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
```

**Comment by Ocramius** [comment]
```
Running before_script: xmllint --schema vendor/squizlabs/php_codesniffer/phpcs.xsd phpcs.xml
warning: failed to load external entity "vendor/squizlabs/php_codesniffer/phpcs.xsd"
Schemas parser error : Failed to locate the main schema resource at 'vendor/squizlabs/php_codesniffer/phpcs.xsd'.
```

This is not caused by your changes, but by the PHPCS version installed being too old.

I think we're good to proceed with this as it is, since upgrading CS rules and tooling is not worth it, on a security-only package, so I'm willing to keep CI red on that specific job.

**Review by Ocramius — APPROVED** [review]
Thanks @glo82145!
